hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/db/RDBBatchOperation.java

```diff
@@ -39,26 +246,38 @@ public RDBBatchOperation(ManagedWriteBatch writeBatch) {
     this.writeBatch = writeBatch;
   }
 
+  @Override
+  public String toString() {
+    return name;
+  }
+
   public void commit(RocksDatabase db) throws IOException {
+    debug(() -> String.format("%s: commit %s",
+        name, opCache.getCommitString()));
+    opCache.prepareBatchWrite();
     db.batchWrite(writeBatch);
   }
 
   public void commit(RocksDatabase db, ManagedWriteOptions writeOptions)
       throws IOException {
+    debug(() -> String.format("%s: commit-with-writeOptions %s",
+        name, opCache.getCommitString()));
+    opCache.prepareBatchWrite();
```

Comment:
I'd just pass `writeBatch` to `opCache.prepareBatchWrite()` to remove the nested dependency. `OpCache` can be considered being outside with a better name, e.g. `BatchOperationCache`.

Reply:
I guess you mean changing the classes to `static`?  I actually start with `static` and passing `writeBatch` around.  The `static` keywords were removed when printing the debug messages with the `name`.  It does not look good to pass also the `name` around.